lib/reservation.rb

```diff
@@ -0,0 +1,12 @@
+class Reservation
+  attr_reader :date_range, :room_num, :rate, :cost
+
+  def initialize(date_range, room_num, rate)
+    @date_range = {}
+    @date_range[:begin_date] = date_range[:begin_date]
+    @date_range[:end_date] = date_range[:end_date]
+    @room_num = room_num
+    @rate = rate
+    @cost = ( date_range[:end_date] - date_range[:begin_date] ) * rate
```

Comment:
I would make `cost` a method here rather than an instance variable. That way if the date range of this reservation changes, the cost will be updated automatically. The downside is you have to do a little more work every time you want the cost, but it's exactly the sort of work computers are really good at.

This is the sort of small thing that makes code easier to change in the future.